zerver/middleware.py

```diff
@@ -519,25 +519,28 @@ def process_request(self, request: HttpRequest) -> Optional[HttpResponse]:
         return None
 
 
-class SetRemoteAddrFromForwardedFor(MiddlewareMixin):
-    """
-    Middleware that sets REMOTE_ADDR based on the HTTP_X_FORWARDED_FOR.
+class SetRemoteAddrFromRealIpHeader(MiddlewareMixin):
+    """Middleware that sets REMOTE_ADDR based on the X-Real-Ip header.
+
+    This middleware is similar to Django's old
+    SetRemoteAddrFromForwardedFor middleware.  We use X-Real-Ip, and
+    not X-Forwarded-For, because the latter is a list of proxies, some
+    number of which are trusted by us, and some of which could be
+    arbitrarily set by the user.  nginx has already parsed which are
+    which, and has set X-Real-Ip to the first one, going right to
+    left, which is untrusted.
+
+    Since we are always deployed behind nginx, we can trust the
+    X-Real-Ip which is so set.
```

Comment:
Should we remark what happens in a development environment where `nginx` isn't present?

Reply:
In a development environment, we just fall back to the remote IP passed in from UWSGI, which is what we've always done.  I can add a note to that effect.

As noted in the commit message, tornado was already using `X-Real-IP`. (e.g. https://github.com/zulip/zulip/blob/master/puppet/zulip/files/nginx/zulip-include-frontend/app#L49)